### DISCUSSION TITLE
Axial Induction formulation in FLORIS

### DISCUSSION THREAD

**Discussion by windexplorer21** [NatLabRockies/floris#1010] [Q&A] [answered]
Hello everyone,

I am trying to understand what formula is implemented in FLORIS for calculating axial induction. I have investigated the `get_turbine_ais()` function but have not been able to grasp how this value is actually calculated.

If anyone could provide clarification or guidance on how FLORIS performs this calculation, I would greatly appreciate it.

Thank you in advance for your help.
windexplorer21

**Comment by misi9170** [comment, marked as answer]
Hi @windexplorer21 ,

Thanks for the question and for trying to take a look at it yourself! The axial induction calculations are buried fairly far down in the code, on [operation_models.py](https://github.com/NREL/floris/blob/main/floris/core/turbine/operation_models.py). 

There, you'll see that there are various different turbine operation models, each of which can implement its own calculation of the axial induction factor in its `axial_induction()` method. For example, [here is the axial induction calculation for the `SimpleTurbine` operation model](https://github.com/NREL/floris/blob/main/floris/core/turbine/operation_models.py#L141-L156). You'll see that the `axial_induction()` method first calls `thrust_coefficient()` to get the thrust coefficient value, and then computes the axial induction factor from that.

Cheers,
Misha

**Comment by windexplorer21** [comment]
Dear @misi9170,

Thank you for the quick reply! I appreciate the guidance on where to find the axial induction calculations in operation_models.py. I'm currently working with a CosineLoss turbine and found exactly what I was looking for in the code.

Thanks again for your help!

Best,
@windexplorer21

**Reply by misi9170** [reply]
Great, glad to help!
